src/textual/pilot.py

```diff
@@ -3,24 +3,24 @@
 import rich.repr
 
 import asyncio
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Generic
 
 if TYPE_CHECKING:
-    from .app import App
+    from .app import App, ReturnType
 
 
 @rich.repr.auto(angular=True)
-class Pilot:
+class Pilot(Generic[ReturnType]):
```

Comment:
The issue in this file is that App is Generic, and mypy complains all over. Adding the type parameter everywhere and making `Pilot` generic on it solves the things, which seems like the right thing.